### PR TITLE
fix(ui): route :tag/:untag/:filter-tag to podcast list buffer directly

### LINKED ISSUE #168: [Bug] :tag/:untag/:filter-tag commands silently fail when podcast list is not the active buffer
## Summary

The `:tag`, `:untag`, and `:filter-tag` minibuffer commands silently do nothing when the podcast list buffer is not the currently active/focused buffer. This is because all three commands route the action through `buffer_manager.current_buffer_mut()`, which returns whichever buffer has focus — and only `PodcastListBuffer` handles `AddTag`/`RemoveTag`/`FilterByTag`. All other buffers silently swallow these actions via `_ => UIAction::None` catch-all match arms.

**Parent**: [Epic] Podcast Organization and Discovery (#91)

## Problem / Background

**Current behavior**: User is viewing an episode list, presses `:` and types `tag tech` + Enter. The minibuffer closes. No tag is added, no badge appears, no error is shown. The command is silently lost.

**Expected behavior**: `:tag tech` should always add a tag to the selected podcast, regardless of which buffer is active. If no podcast is determinable from context, show an error message.

**Root cause** — three identical routing bugs in `src/ui/app.rs`:

### 1. `:tag` command (`app.rs:2462–2489`)
```rust
"tag" => {
    let tag = parts[1..].join(" ");
    if let Some(current_buffer) = self.buffer_manager.current_buffer_mut() {
        let result = current_buffer.handle_action(UIAction::AddTag { tag });
        //          ^^^^^^^^^^^^^^^ THIS is the active buffer, NOT necessarily the podcast list
        match result {
            UIAction::TriggerAddTag { .. } => { /* success path */ }
            _ => {}  // ← UIAction::None from non-podcast buffers lands HERE — silently dropped
        }
    }
}
```

### 2. `:untag` command (`app.rs:2490–2517`) — identical pattern
```rust
if let Some(current_buffer) = self.buffer_manager.current_buffer_mut() {
    let result = current_buffer.handle_action(UIAction::RemoveTag { tag });
    // Same bug: routes to active buffer, not podcast list
}
```

### 3. `:filter-tag` command (`app.rs:2544–2558`) — identical pattern
```rust
if let Some(current_buffer) = self.buffer_manager.current_buffer_mut() {
    let result = current_buffer.handle_action(UIAction::FilterByTag { tag });
    // Same bug: routes to active buffer, not podcast list
}
```

**Why other buffers silently swallow**: Every non-podcast-list buffer has a catch-all in `handle_action`:
- `episode_list.rs`: `_ => UIAction::None`
- `episode_detail.rs`: `_ => UIAction::None`
- `whats_new.rs`: `_ => UIAction::None`
- `help.rs`: `_ => UIAction::None`
- `downloads.rs`: `_ => UIAction::None`
- `sync.rs`: `_ => UIAction::None`

Then back in `app.rs`, the `_ => {}` catch-all in the response match silently discards the `UIAction::None` with no user feedback.

**Contrast with `:tags` command** (`app.rs:2518–2543`) which is implemented CORRECTLY — it uses the typed getter `self.buffer_manager.get_podcast_list_buffer_mut()` instead of `current_buffer_mut()`:
```rust
"tags" => {
    let tags: Vec<String> = if let Some(podcast_buffer) =
        self.buffer_manager.get_podcast_list_buffer_mut()  // ← CORRECT: typed getter
    { ... };
}
```

## Proposed Solution

Route `:tag`, `:untag`, and `:filter-tag` directly to the podcast list buffer using the existing typed getter `buffer_manager.get_podcast_list_buffer_mut()` (defined at `buffers/mod.rs:368`), matching the pattern already used by `:tags`.

### Fix for `:tag` (`app.rs:2462`)

```rust
"tag" => {
    if parts.len() > 1 {
        let tag = parts[1..].join(" ");
        if let Some(podcast_buffer) = self.buffer_manager.get_podcast_list_buffer_mut() {
            let result = podcast_buffer.handle_action(UIAction::AddTag { tag });
            match result {
                UIAction::TriggerAddTag { podcast_id, podcast_title, tag } => {
                    self.show_message(format!("Tagged '{}' with \"{}\"", podcast_title, tag));
                    self.trigger_async_add_tag(podcast_id, tag);
                }
                UIAction::ShowMessage(msg) => self.show_message(msg),
                UIAction::ShowError(msg) => self.show_error(msg),
                _ => {
                    self.show_error("Could not add tag. Is a podcast selected?".to_string());
                }
            }
        } else {
            self.show_error("Podcast list not available".to_string());
        }
        Ok(true)
    } else { ... }
}
```

### Fix for `:untag` (`app.rs:2490`)

Same pattern — replace `self.buffer_manager.current_buffer_mut()` with `self.buffer_manager.get_podcast_list_buffer_mut()`.

### Fix for `:filter-tag` (`app.rs:2544`)

Same pattern — replace `self.buffer_manager.current_buffer_mut()` with `self.buffer_manager.get_podcast_list_buffer_mut()`.

**Additionally**, for `:filter-tag`, consider switching to the podcast list buffer after applying the filter so the user can see the filtered result:
```rust
"filter-tag" => {
    if parts.len() > 1 {
        let tag = parts[1..].join(" ");
        if let Some(podcast_buffer) = self.buffer_manager.get_podcast_list_buffer_mut() {
            let result = podcast_buffer.handle_action(UIAction::FilterByTag { tag: tag.clone() });
            // ... handle result ...
        }
        // Switch to podcast list so user sees the filter applied
        let _ = self.buffer_manager.switch_to_buffer(&"podcast-list".to_string());
        Ok(true)
    }
}
```

**Why not add AddTag handling to all buffers?** Tags are a podcast-level concept. Adding `AddTag` handling to episode_list, help, downloads, etc. would require each buffer to somehow know the "parent podcast" context, which is architecturally wrong. The command should go directly to the podcast list buffer where the selection and podcast data lives.

## Files to Modify

| File | Change |
|------|--------|
| `src/ui/app.rs:2462–2489` | `:tag` — replace `current_buffer_mut()` with `get_podcast_list_buffer_mut()`, add fallback error |
| `src/ui/app.rs:2490–2517` | `:untag` — same fix |
| `src/ui/app.rs:2544–2558` | `:filter-tag` — same fix, optionally switch to podcast list |

## Acceptance Criteria

- [ ] `:tag tech` adds the tag to the selected podcast **regardless of which buffer is active**
- [ ] `:untag tech` removes the tag **regardless of which buffer is active**
- [ ] `:filter-tag tech` filters the podcast list **regardless of which buffer is active**
- [ ] When no podcast is selected in the podcast list, `:tag tech` shows a clear error message (not silence)
- [ ] When the podcast list buffer doesn't exist (edge case), a clear error message is shown
- [ ] The `[tech]` badge appears in the podcast list immediately after tagging (optimistic update still works)
- [ ] The `:tags` command continues to work (no regression — it already uses the correct getter)
- [ ] All existing tests pass (`cargo test`)
- [ ] `cargo clippy -- -D warnings` passes
- [ ] `cargo fmt --check` passes

### Unit Tests
- `test_tag_command_routes_to_podcast_list_not_active_buffer`: Set up app with episode list as active buffer, dispatch `:tag tech`, verify the podcast list buffer received the tag (not the episode list)
- `test_untag_command_routes_to_podcast_list`: Same pattern for `:untag`
- `test_filter_tag_routes_to_podcast_list`: Same pattern for `:filter-tag`
- `test_tag_command_shows_error_when_no_podcast_selected`: No podcast selected in podcast list, verify error message
- `test_tag_command_shows_error_when_podcast_list_unavailable`: Edge case coverage

### Manual Tests
1. Open an episode list (Enter on a podcast) → `:tag tech` → switch back to podcast list (F2) → `[tech]` badge should be visible
2. Open help buffer (F1) → `:tag news` → switch to podcast list → `[news]` badge visible
3. Open downloads (F4) → `:untag news` → switch to podcast list → `[news]` badge gone
4. Open episode list → `:filter-tag tech` → podcast list should show only tagged podcasts

## Implementation Notes

- The `get_podcast_list_buffer_mut()` typed getter at `buffers/mod.rs:368` returns `Option<&mut PodcastListBuffer>`. It uses safe downcasting via `as_any_mut().downcast_mut()`. This already exists and is battle-tested (used by `:tags`, podcast refresh, etc.).
- The optimistic in-memory update in `PodcastListBuffer::handle_action(AddTag)` mutates `self.podcasts[idx]` directly — this still works correctly when called via the typed getter.
- Consider a follow-up (out of scope) to audit other podcast-scoped commands (`:filter-status`, `:sort`, etc.) for the same routing bug pattern.

## Definition of Done

- [ ] Functionality implemented according to requirements
- [ ] Unit tests written and passing
- [ ] Code passes linting (clippy) with no warnings
- [ ] Code formatted with rustfmt
- [ ] No documentation changes needed (commands already documented correctly)

### PR DESCRIPTION
**Pull request by lqdev**
## Summary

Closes #168 — `:tag`, `:untag`, and `:filter-tag` minibuffer commands silently did nothing when any buffer other than the podcast list had focus. All three commands used `current_buffer_mut()`, routing the action to whichever buffer was active. Non-podcast-list buffers return `UIAction::None` from their catch-all match arm, which was then silently discarded.

## Changes

- `src/ui/app.rs`: Three routing fixes — `:tag`, `:untag`, and `:filter-tag` each replaced `self.buffer_manager.current_buffer_mut()` with `self.buffer_manager.get_podcast_list_buffer_mut()` (the same typed getter already used correctly by `:tags`)
  - `:tag`: now routes to podcast list; catch-all changed to show "Could not add tag. Is a podcast selected?" for unexpected variants; added `else` branch showing "Podcast list not available"
  - `:untag`: same pattern as `:tag`
  - `:filter-tag`: same routing fix; after applying the filter, automatically switches to the podcast-list buffer so the user sees the filtered result immediately
- `CHANGELOG.md`: Added entry under `### Fixed` in `[Unreleased]`

## Manual Testing

**Before (broken behavior):**
1. `cargo run`
2. Select a podcast in the podcast list and press `Enter` to open its episode list
3. Press `:` and type `tag tech` + Enter — minibuffer closes silently, no tag applied, no error shown

**After (fixed behavior):**
1. `cargo run`
2. Select a podcast and press `Enter` to open its episode list
3. Press `:` and type `tag tech` + Enter — podcast is tagged; switch back to podcast list (press `F2`) — `[tech]` badge visible next to the podcast title
4. From episode list, type `:untag tech` + Enter — tag removed (verify in podcast list)
5. From episode list, type `:filter-tag tech` — podcast list buffer opens automatically, showing only tagged podcasts
6. From episode list with no podcast selected in podcast list, type `:tag news` — error message shown in minibuffer: "No podcast selected" (not silence)

## Tests

5 new unit tests:
- `test_tag_command_routes_to_podcast_list_not_active_buffer` — episode list active; verify tag applied to podcast list buffer
- `test_untag_command_routes_to_podcast_list_not_active_buffer` — same pattern for `:untag`
- `test_filter_tag_routes_to_podcast_list_and_switches_buffer` — verifies filter applied and buffer switches to podcast-list
- `test_tag_command_shows_error_when_no_podcast_selected` — empty podcast list; verifies minibuffer shows error (no silent failure)
- `test_untag_command_shows_error_when_no_podcast_selected` — same for `:untag`

578 total unit tests pass (was 573).

## Quality

- `cargo fmt --check` ✅
- `cargo clippy -- -D warnings` ✅
- `cargo test` ✅ (578 passed; pre-existing `test_parse_local_opml` failure is unrelated)

### CODE REVIEW

File: src/ui/app.rs [hunk 1]
```diff
@@ -2655,17 +2665,22 @@ impl UIApp {
...
                             _ => {}
                         }
                     } else {
-                        self.show_error("No buffer available for filtering".to_string());
+                        self.show_error("Podcast list not available".to_string());
                     }
+                    // Switch to podcast list so user sees the filter applied
+                    let _ = self
+                        .buffer_manager
+                        .switch_to_buffer(&"podcast-list".to_string());
```
[the remark this answers is not in the document]
Comment (lqdev) [reply src/ui/app.rs]: Accepted. Moved switch_to_buffer inside the if let Some(podcast_buffer) block (commit 069b2ae) so it only executes when the podcast list buffer exists and the filter was successfully applied. The unconditional call was a bug — in the lse branch it would silently attempt to switch to a buffer that doesn't exist.
